src/components/tasks/card/AssigneeDropdownMenu.tsx

```diff
@@ -0,0 +1,86 @@
+import { FC, useState, useEffect, useMemo } from "react";
+
+import userType from '@/interfaces/user.type';
+import { ALL_USERS } from '@/components/constants/url';
+import fetch from '@/helperFunctions/fetch';
+import { toast, ToastTypes } from '@/helperFunctions/toast';
+
+import classNames from '@/components/tasks/card/card.module.scss';
+
+type Props = {
+  updateAssignee: (username: string) => void
+}
+
+//https://github.com/Real-Dev-Squad/website-status/issues/303
+
+const AssigneeDropdownMenu: FC<Props> = ({ updateAssignee }) => {
+  const [searchValue, setSearchValue] = useState("");
+  const [filteredList, setFilteredList] = useState([]);
+  const [assigneeList, setAssigneeList] = useState([])
+  const { ERROR } = ToastTypes;
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const { requestPromise } = fetch({ url: ALL_USERS });
+        const { data } = await requestPromise;
+        const usersData = data.members || data.users
+        setAssigneeList(usersData)
+      } catch (err: any) {
+        toast(ERROR, err.message);
+      }
+    };
+    fetchData();
+
```

Comment:
is it a good way to call an async method in useEffect?

Reply:
I want to make a fetch call only when this particular component mounts, please advise if there's any other way to do it

Reply:
@harshith-venkatesh would be a better person to contact about this. I guess you need to remove the async function out of use effect and just use the function name in useEffect

Reply:
okay

Reply:
Both approaches are perfectly fine. 
In case you wish to move the function out of `useEffect`, and call it inside `useEffect`, then you have to put the `function` inside dependency array.

Reply:
What @rajat-mehra05 mentioned is fine, however you need to look into dependency and add them accordingly in dependency array.
It's good to take async function outside rather.
But the way you have called works as it returns a value/promise

Reply:
Okay thankyou